### ISSUE TITLE
Unhelpful error message with MutuallyExclusivePropertiesException

### ISSUE THREAD

**Issue by rolweber** [CycloneDX/cyclonedx-python#461]
I've installed cyclonedx-bom into a conda environment with 300+ packages from conda and pip. Running...
```bash
cyclonedx-bom -e --format json -o -
```
results in a stack trace...
```txt
Traceback (most recent call last):
  File "/opt/conda/envs/Python-3.10/bin/cyclonedx-bom", line 8, in <module>
    sys.exit(main())
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/client.py", line 307, in main
    CycloneDxCmd(args).execute()
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/client.py", line 147, in execute
    output = self.get_output()
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/client.py", line 84, in get_output
    parser = self._get_input_parser()
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/client.py", line 254, in _get_input_parser
    return EnvironmentParser(use_purl_bom_ref=self._arguments.use_purl_bom_ref)
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/parser/environment.py", line 86, in __init__
    c.licenses.add(LicenseChoice(license_=License(
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx/model/__init__.py", line 554, in __init__
    raise MutuallyExclusivePropertiesException('Either `spdx_license_id` or `license_name` MUST be supplied')
cyclonedx.exception.model.MutuallyExclusivePropertiesException: Either `spdx_license_id` or `license_name` MUST be supplied
```
I guess that (at least) one of the installed packages has non-compliant license info.

1. Could you please add the name of the offending package to the error message?
2. Wouldn't it make more sense to continue processing the remaining packages?
   - The tool could generate a full list of packages with problems before bailing out.
   - The tool could work for the other packages. Maybe with an extra option `--tolerate-errors-in-license-info` or similar.

**Comment by rolweber** [comment]
Output with `-X`:
```txt
[DEBUG] - 2022-12-12 12:42:02.586742 - !!! DEBUG MODE ENABLED !!!
[DEBUG] - 2022-12-12 12:42:02.586798 - Parsed Arguments: Namespace(input_from_conda_explicit=False, input_from_conda_json=False, input_from_environment=True, input_from_poetry=False, input_from_pip=False, input_from_requirements=False, input_source=None, output_format='json', output_schema_version='1.4', output_file='-', output_file_overwrite=False, use_purl_bom_ref=False, debug_enabled=True)
[DEBUG] - 2022-12-12 12:42:02.586816 - output_format: _CLI_OUTPUT_FORMAT.JSON
Traceback (most recent call last):
  File "/opt/conda/envs/Python-3.10/bin/cyclonedx-bom", line 8, in <module>
    sys.exit(main())
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/client.py", line 307, in main
    CycloneDxCmd(args).execute()
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/client.py", line 147, in execute
    output = self.get_output()
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/client.py", line 84, in get_output
    parser = self._get_input_parser()
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/client.py", line 254, in _get_input_parser
    return EnvironmentParser(use_purl_bom_ref=self._arguments.use_purl_bom_ref)
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx_py/parser/environment.py", line 86, in __init__
    c.licenses.add(LicenseChoice(license_=License(
  File "/opt/conda/envs/Python-3.10/lib/python3.10/site-packages/cyclonedx/model/__init__.py", line 554, in __init__
    raise MutuallyExclusivePropertiesException('Either `spdx_license_id` or `license_name` MUST be supplied')
cyclonedx.exception.model.MutuallyExclusivePropertiesException: Either `spdx_license_id` or `license_name` MUST be supplied
```

**Comment by jkowalleck** [comment]
a hotfix was released via [v3.7.4](https://github.com/CycloneDX/cyclonedx-python/releases/tag/v3.7.4).

The entire story is not done.
I am currently working on enhancements of the debuggability.

**Comment by rolweber** [comment]
Thank you!

The problem no longer occurs. Tested with the following packages:
```txt
Successfully installed cyclonedx-bom-3.8.0 cyclonedx-python-lib-3.1.1 packageurl-python-0.10.4 pip-requirements-parser-32.0.0 toml-0.10.2
```
